src/components/MapArea.js

```diff
@@ -40,17 +57,34 @@ export default function MapArea() {
 	useEffect(() => {    
 		// create the map and configure it    
 		// check out the API reference for more options    
-		// https://docs.mapbox.com/mapbox-gl-js/api/map/  
-    // if (map.current) return;  
 		map.current = new mapboxgl.Map({      
 			container: mapContainer.current,      
 			style,      
 			center: [longitude, latitude],      
 			zoom,    
 		})
 
-		// cleanup function to remove map on unmount
+    map.current.on('load', function() {
+      map.current.addSource('trailData', {
+        'type': 'geojson',
+        'data': trailData
+      })
+  
+      map.current.addLayer({
+        'id': 'trailData',
+        'type': 'line',
+        'source': 'trailData',
+        'layout': {},
+        'paint': {
+          'line-color': '#ec2222',
+          'line-width': 2,
+          'line-dasharray': [2, 1]
+        }
+      })
+    })
+
 		return () => map.current.remove() 
+
 	}, [style])
```

Comment:
You mention that you have an empty dependency array so that the map is only created once (which is correct), but here you have added `style` as a dependency. The way this component is switching styles currently, is by calling `useEffect` everytime you change the `style` state, which ends up re-creating the map instance, instead of just changing the "baselayer". 

We have a ref to the map object, so we have access to the function `setStyle` through `map.current.setStyle()`. This is the function that should be used to switch styles for a map, instead of re-creating the map instance.